.github/workflows/build_and_release.yml

```diff
@@ -0,0 +1,85 @@
+name: 📦 Build and Release
+
+on:
+  push:
+    paths:
+      - lib/justifi/version.rb
+
+jobs:
+  run_tests:
+    name: 📋 Verify
+    uses: justifi-tech/justifi-ruby/.github/workflows/verify.yml@main
+    if: github.ref == 'refs/heads/main'
+  build_and_release:
+    strategy:
+      matrix:
+        include:
+          - api_key: JUSTIFI_MACHINA_RUBYGEMS_API_KEY
+            bearer: false
+            gem_host: https://rubygems.org
+            pkg_host_name: rubygems.org
+          - api_key: JUSTIFI_MACHINA_GITHUB_TOKEN
+            bearer: true
+            gem_host: https://rubygems.pkg.github.com/${{ github.repository_owner }}
+            pkg_host_name: pkg.github.com
+
+    name: 📦 Package and Release on ${{ matrix.pkg_host_name }}
+    needs: run_tests
+    if: github.ref == 'refs/heads/main'
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Build API_KEY value
+        id: api_key
+        run: |
+          echo "::set-output name=value::${{ matrix.bearer && 'Bearer ' || '' }}${{ secrets[matrix.api_key] }}"
```

Comment:
Things got a little funky here because we are dual publishing, and you can't pass `${{ secrets.MY_SECRET }}` values in the `matrix`.

However, you can say the secret name, and use the `${{ secrets[MY_SECRET] }}` notation to look up a secret, so that is what is done here.

Also, since only the `rubygems.pkg.github.com` host has a `Bearer ...` prefix, a bit of extra logic is here for that.  I double checked that the key is still masked when doing this, and it is (even removes the `Bearer ` part from the logs, which is fine.

Finally... I am doing this as a separate step for if in the future, we want this as a shared action, this would be the only step that is needed, and the rest can be modularized and shared.